src/Symfony/Component/Workflow/SupportStrategy/ClassInstanceSupportStrategy.php

```diff
@@ -1,13 +1,22 @@
 <?php
 
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
 namespace Symfony\Component\Workflow\SupportStrategy;
 
-use Symfony\Component\Workflow\Workflow;
+use Symfony\Component\Workflow\WorkflowInterface;
 
 /**
  * @author Andreas Kleemann <akleemann@inviqa.com>
  */
-final class ClassInstanceSupportStrategy implements SupportStrategyInterface
+final class ClassInstanceSupportStrategy implements SupportStrategyWorkflowInterface
```

Comment:
Not implementing the old interface is a BC break (`$strategy instanceof SupportsStrategyInterface` returning false). Personally I wouldn't care, this could be internal

Reply:
I think this is internal too.

Reply:
Internal should be hinted by an annotation, we can't have a policy of telling "that's internal" and break BC.
I suggest to open a PR against 3.4 to really mark them internal so that we're free to change in 4.1.

Reply:
Ill open that PR.